src/test/java/com/google/firebase/auth/FirebaseAuthIT.java

```diff
@@ -1066,6 +1066,137 @@ public void testTenantAwareOidcProviderConfigLifecycle() throws Exception {
+            .setClientId("CLIENT_ID")
+            .setIssuer("https://oidc.com/issuer");
+        auth.createOidcProviderConfig(createRequest);
+      }
+
+      // Test list by batches
+      final AtomicInteger collected = new AtomicInteger(0);
+      ListProviderConfigsPage<OidcProviderConfig> page =
+          auth.listOidcProviderConfigsAsync(null).get();
+      while (page != null) {
+        for (OidcProviderConfig providerConfig : page.getValues()) {
+          if (providerIds.contains(providerConfig.getProviderId())) {
+            collected.incrementAndGet();
+            assertEquals("CLIENT_ID", providerConfig.getClientId());
+            assertEquals("https://oidc.com/issuer", providerConfig.getIssuer());
+          }
+        }
+        page = page.getNextPage();
+      }
+      assertEquals(providerIds.size(), collected.get());
+
+      // Test iterate all
+      collected.set(0);
+      page = auth.listOidcProviderConfigsAsync(null).get();
+      for (OidcProviderConfig providerConfig : page.iterateAll()) {
+        if (providerIds.contains(providerConfig.getProviderId())) {
+          collected.incrementAndGet();
+          assertEquals("CLIENT_ID", providerConfig.getClientId());
+          assertEquals("https://oidc.com/issuer", providerConfig.getIssuer());
+        }
+      }
+      assertEquals(providerIds.size(), collected.get());
+
+      // Test iterate async
+      collected.set(0);
+      final Semaphore semaphore = new Semaphore(0);
+      final AtomicReference<Throwable> error = new AtomicReference<>();
+      ApiFuture<ListProviderConfigsPage<OidcProviderConfig>> pageFuture =
+          auth.listOidcProviderConfigsAsync(null);
+      ApiFutures.addCallback(
+          pageFuture,
+          new ApiFutureCallback<ListProviderConfigsPage<OidcProviderConfig>>() {
+            @Override
+            public void onFailure(Throwable t) {
+              error.set(t);
+              semaphore.release();
+            }
+
+            @Override
+            public void onSuccess(ListProviderConfigsPage<OidcProviderConfig> result) {
+              for (OidcProviderConfig providerConfig : result.iterateAll()) {
+                if (providerIds.contains(providerConfig.getProviderId())) {
+                  collected.incrementAndGet();
+                  assertEquals("CLIENT_ID", providerConfig.getClientId());
+                  assertEquals("https://oidc.com/issuer", providerConfig.getIssuer());
+                }
+              }
+              semaphore.release();
+            }
+          }, MoreExecutors.directExecutor());
+      semaphore.acquire();
+      assertEquals(providerIds.size(), collected.get());
+      assertNull(error.get());
+    } finally {
+      // Delete provider configs
+      for (String providerId : providerIds) {
+        auth.deleteProviderConfigAsync(providerId).get();
+      }
+    }
+  }
+
+  @Test
+  public void testTenantAwareListOidcProviderConfigs() throws Exception {
```

Comment:
I feel like we are creating so many throw away tenants in these tests. I'd suggest that we move all tenant-aware tests to a new class. We can just create one test tenant in a `@BeforeClass` and reuse it as much as possible. Will make the overall logic a lot clear that way too.

But feel free to do that in a separate PR.

Reply:
This is a really good idea! I think this would clean things up a lot. I've added a TODO to address in a separate PR (since it will involve moving quite a bit of code).